src/cf.data.pre

```diff
@@ -1071,8 +1071,9 @@ DEFAULT: all src all
 DEFAULT: manager url_regex -i ^cache_object:// +i ^[^:]+://[^/]+/squid-internal-mgr/
 DEFAULT: localhost src 127.0.0.1/32 ::1
 DEFAULT: to_localhost dst 127.0.0.0/8 0.0.0.0/32 ::1/128 ::/128
+DEFAULT: to_linklocal dst 169.254.0.0/16 fe80::/10
```

Comment:
> Amos at https://github.com/squid-cache/squid/pull/1161#issuecomment-1269266477: IMO the same rationale should be used for site-local connections. That is the range FEC0::/10.

I disagree that [site-local](https://4sysops.com/archives/ipv6-tutorial-part-6-site-local-addresses-and-link-local-addresses/) addresses fall into the same category as the "localhost and equivalent" addresses this PR proposes to block by default. AFAICT, site-local are used primarily for communicating with other hosts (within the same organization). The proposed addition is not about that kind of communication. In other words, the proposed addition is not about "local to the organization" in general, but about "same-host" specifically.

Once could argue that denied link-local addresses are not _necessarily_ same-host either. That is a valid argument, but we are explicitly making an _exception_ there because of the widespread use of those addresses for a specific _same-host_ use case: sensitive instance metaservices in the cloud[^1]. Site-local addresses do not fall into this specific exception. They are also [deprecated](https://www.ietf.org/rfc/rfc3879.txt) since 2004. Overall, adding site-local would create more confusion about the intent of these rules without covering many actual use cases.

[^1]: If we had a more precise way to cover that common use case without creating a list of popular cloud services, we would have used that instead of to_linklocal.

Reply:
I meant the same reasons apply to justify considering a ```to_sitelocal``` or ```to_localnet``` ACL separate to this ```to_linklocal```. **IF** we have any reason to add such a ```dst``` ACL.

The central consideration for all these is the security restriction preventing Squid routing WAN traffic into sensitive scopes. In this regard Squid should comply with and enforce (by default) the same behaviour/limits the RFCs require routers to use.

Reply:
> I meant the same reasons apply to justify considering a `to_sitelocal` or `to_localnet` ACL separate to this `to_linklocal`. **IF** we have any reason to add such a `dst` ACL.

> The central consideration for all these is the security restriction preventing Squid routing WAN traffic into sensitive scopes. In this regard Squid should comply with and enforce (by default) the same behaviour/limits the RFCs require routers to use.

AFAICT, you want to enlarge the default protection scope from PR's "localhost and equivalent" to something like "local net" (comprising all non-routable addresses?)[^1]. In principle, I am _not_ against that or any enlargement -- we just need to clearly define the new scope, provide a rationale for it, and match it with a new set of ACLs (without creating more problems that the new default would solve).

I do not know what RFCs say about default router configurations, but I believe we must balance purely technical security needs (prescribing a simple `http_access deny all` default) with human tendencies to remove obstacles instead of fine-tuning them (making a simple `http_access deny all` default a terrible idea). Improving that balance requires more work than we have done so far in this PR. Most likely, we need to do more than adding a couple of configuration lines and quickly writing a paragraph of text.

To make steady progress, I plan to merge this PR and welcome new/followup PR(s) that properly enlarge protection scope further, including PRs that make (relatively narrow rules added in) this PR unnecessary.

[^1]: Or something like that -- unfortunately, I do not know what addresses routers are supposed to protect by default according to RFCs.